imsim/skycat.py

```diff
@@ -0,0 +1,355 @@
+        return internal_av, internal_rv, galactic_av, galactic_rv
+
+    def get_gsobject(self, skycat_obj, component, gsparams, rng, exp_time):
+        """
+        Return a galsim.GSObject for the specifed skyCatalogs object
+        and component.
+
+        Parameters
+        ----------
+        skycat_obj : skyCatalogs.BaseObject
+            A skyCatalogs object, e.g., a star or galaxy.
+        component : str
+            The name of the sub-component of the skyCatalogs object
+            to consider.
+        """
+        sed, magnorm = self.getSED_info(skycat_obj, component)
+        if sed is None or magnorm >= 50:
+            return None
+
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+
+        if skycat_obj.object_type == 'star':
+            obj = galsim.DeltaFunction(gsparams=gsparams)
+        elif (skycat_obj.object_type == 'galaxy' and
+              component in ('bulge', 'disk', 'knots')):
+            my_component = component
+            if my_component == 'knots':
+                my_component = 'disk'
+            a = skycat_obj.get_native_attribute(f'size_{my_component}_true')
+            b = skycat_obj.get_native_attribute(f'size_minor_{my_component}_true')
+            assert a >= b
+            pa = skycat_obj.get_native_attribute('position_angle_unlensed')
+            beta = float(90 + pa)*galsim.degrees
+            hlr = (a*b)**0.5   # approximation for half-light radius
+            if component == 'knots':
+                npoints = skycat_obj.get_native_attribute('n_knots')
+                assert npoints > 0
+                obj =  galsim.RandomKnots(npoints=npoints,
+                                          half_light_radius=hlr, rng=rng,
+                                          gsparams=gsparams)
+            else:
+                n = skycat_obj.get_native_attribute(f'sersic_{component}')
+                # Quantize the n values at 0.05 so that galsim can
+                # possibly amortize sersic calculations from the previous
+                # galaxy.
+                n = round(n*20.)/20.
+                obj = galsim.Sersic(n=n, half_light_radius=hlr,
+                                    gsparams=gsparams)
+            shear = galsim.Shear(q=b/a, beta=beta)
+            obj = obj._shear(shear)
+            g1, g2, mu = self.getLens(skycat_obj)
+            obj = obj._lens(g1, g2, mu)
+        else:
+            raise RuntimeError("Do not know how to handle object type: %s" %
+                               component)
+
+        # The seds are normalized to correspond to magnorm=0.
+        # The flux for the given magnorm is 10**(-0.4*magnorm)
+        # The constant here, 0.9210340371976184 = 0.4 * log(10)
+        flux = math.exp(-0.9210340371976184 * magnorm)
+
+        # This gives the normalization in photons/cm^2/sec.
+        # Multiply by area and exptime to get photons.
+        fAt = flux * self._rubin_area * exp_time
```

Comment:
Actually, here is the right place I think.  This is the relevant flux.  We can then even do the poisson realization of it here and do `withFlux` with that value.  And if the realized flux is what below we call faint, use the trivial sed rather than making the right one.

But that means that we need to split up the get_SEDinfo method to have a fast way to get magnorm first that doesn't do any complex calculations (hopefully none are done internally in skyCatalog to get magnorm), and then only if we care about the real SED do the full calculation with redshift and extinction and all that.